lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1596,14 +1596,15 @@ createDispatchingDiagnosticConsumerIfNeeded(
   // So a value of "4" here means that there would be no heap allocation on a
   // clean build of a module with up to 32 files on an 8-core machine, if the
   // user doesn't customize anything.
-  SmallVector<FileSpecificDiagnosticConsumer::ConsumerPair, 4> subConsumers;
+  SmallVector<FileSpecificDiagnosticConsumer::Subconsumer, 4> subconsumers;
 
   inputsAndOutputs.forEachInputProducingSupplementaryOutput(
       [&](const InputFile &input) -> bool {
-    if (auto subConsumer = maybeCreateSingleConsumer(input))
-      subConsumers.emplace_back(input.file(), std::move(subConsumer));
-    return false;
-  });
+        if (auto subconsumer = maybeCreateSingleConsumer(input))
+          subconsumers.emplace_back(FileSpecificDiagnosticConsumer::Subconsumer(
```

Comment:
This is `emplace_back`, not `push_back`, precisely so that you don't have to name the type again.

Reply:
Like magic!